cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/RuntimeClientService.java

```diff
@@ -201,13 +252,14 @@ private class TopicRelayer implements Closeable {
         LogSamplers.limitRate(TimeUnit.SECONDS.toMillis(30)));
 
     protected final TopicId topicId;
+    private final MessageAggregator messageAggregator;
     private String lastMessageId;
     private long nextPublishTimeMillis;
     private int totalPublished;
 
-
-    TopicRelayer(TopicId topicId) {
+    TopicRelayer(TopicId topicId, @Nullable MessageAggregator messageAggregator) {
```

Comment:
Preferably not to have so many `@Nullable` field. This is an indicator that a better encapsulation and abstraction is needed (hence refactoring).